File: Factory/SimpleFactory/Java/src/Pizza/PepperoniPizza.java
package eu.svetlin.design.patterns.factory.simplefactory.Pizza;

public class PepperoniPizza extends Pizza {
    public PepperoniPizza() {
        name = "Pepperoni Pizza";
        dough = "Regular Crust";
        sauce = "Marinara Sauce";
        toppings.add("Sliced Pepperoni");
        toppings.add("Sliced Onion");
        toppings.add("Grated Parmesan Cheese");
    }
}

File: Factory/SimpleFactory/TypeScript/src/Pizza/PepperoniPizza.ts
import Pizza from './Pizza';

class PepperoniPizza extends Pizza {
  constructor() {
    super();
    this.name = 'Pepperoni Pizza';
    this.dough = 'Regular Crust';
    this.sauce = 'Marinara Sauce';
    this.toppings.push('Sliced Pepperoni');
    this.toppings.push('Sliced Onion');
    this.toppings.push('Grated Parmesan Cheese');
  }
}

export default PepperoniPizza;
